validator/db/kv/db.go

```diff
@@ -107,9 +116,18 @@ func NewKVStore(dirPath string, pubKeys [][48]byte) (*Store, error) {
 		}
 	}
 
-	err = prometheus.Register(createBoltCollector(kv.db))
-
-	return kv, err
+	// We then fetch the attestation histories for each public key
+	// and store them in a map for usage at runtime.
+	if !featureconfig.Get().DisableAttestingHistoryDBCache {
+		for _, pubKey := range pubKeys {
+			history, err := kv.AttestationHistoryForPubKeyV2(ctx, pubKey)
+			if err != nil {
+				return nil, err
+			}
+			kv.attestingHistoriesByPubKey[pubKey] = history
```

Comment:
I know this is `NewKVStore`, but a locking here makes me feel safer

Reply:
Discussed offline that we cant' use a lock here because the function `AttestationHistoryForPubKey()` will use the lock as well.

Let's add a comment to clarify this.